functions/Start-DbaAgentJob.ps1

```diff
@@ -120,69 +164,72 @@ function Start-DbaAgentJob {
+                    Write-Message -Level Verbose -Message "Job $($currentjob.Name) status is $($currentjob.CurrentRunStatus)"
+                    Write-Message -Level Verbose -Message "Job $($currentjob.Name) last run date is $($currentjob.LastRunDate)"
 
-                        Write-Message -Level Verbose -Message "Sleeping for $SleepPeriod ms and refreshing"
-                        Start-Sleep -Milliseconds $SleepPeriod
-                        $currentjob.Refresh()
+                    Write-Message -Level Verbose -Message "Sleeping for $SleepPeriod ms and refreshing"
+                    Start-Sleep -Milliseconds $SleepPeriod
+                    $currentjob.Refresh()
 
-                        # If it failed fast, speed up output
-                        if ($lastrun -ne $currentjob.LastRunDate) {
-                            $i = 600
-                        }
+                    # If it failed fast, speed up output
+                    if ($lastrun -ne $currentjob.LastRunDate) {
+                        $i = 600
                     }
+                }
 
-                    # Wait for the job
-                    if (Test-Bound -ParameterName Wait) {
-                        while ($currentjob.CurrentRunStatus -ne 'Idle') {
-                            $currentRunStatus = $currentjob.CurrentRunStatus
-                            $currentStep = $currentjob.CurrentRunStep
-                            $jobStepsCount = $currentjob.JobSteps.Count
-                            $currentStepRetryAttempts = $currentjob.CurrentRunRetryAttempt
-                            if (-not $currentStepRetryAttempts) { $currentStepRetryAttempts = "0" }
-                            $currentStepRetries = $currentjob.RetryAttempts
-                            if (-not $currentStepRetries) { $currentStepRetries = "Unknown" }
-                            Write-Message -Level Verbose -Message "$currentjob is $currentRunStatus, currently on Job Step $currentStep / $jobStepsCount, and has tried $currentStepRetryAttempts / $currentStepRetries retry attempts"
-                            Start-Sleep -Seconds $WaitPeriod
-                            $currentjob.Refresh()
-                        }
-                        Get-DbaAgentJob -SqlInstance $server -Job $currentjob.Name
-                    } else {
-                        Get-DbaAgentJob -SqlInstance $server -Job $currentjob.Name
+                if ((Test-Bound -ParameterName Wait) -and (Test-Bound -not -ParameterName Parallel)) {
+                    # Wait for the job each job in a serialized fashion.
+                    while ($currentjob.CurrentRunStatus -ne 'Idle') {
+                        Invoke-Command -ScriptBlock $waitBlock -ArgumentList @($currentjob, $true, $WaitPeriod)
                     }
+                    Get-DbaAgentJob -SqlInstance $server -Job $($currentjob.Name)
+                } elseif (Test-Bound -not -ParameterName Parallel) {
```

Comment:
Same idea here. Test-Bound is just validation if the parameter is provided, not its value. As I understand, this was the way it was used before, and this is an opportunity to fix it.